### PR TITLE
Merge in Components from Blazor repo

### PR DESCRIPTION
**Pull request by SteveSandersonMS**
As discussed with @natemcmaster and @ryanbrandenburg yesterday, this brings in the parts of the Blazor repo that we want.

By branch-filtering out the Mono binaries, the size of the added content is reduced from ~250MB to ~11MB. Doing this means all the historical Blazor commit SHAs have changed, but we think it's worth it to keep the mondorepo size down. We are still retaining other aspects of the Blazor history (individual commits, authors, etc) so the important value is retained.

### CODE REVIEW

File: build/buildorder.props [hunk 1]
```diff
@@ -29,6 +29,7 @@
     <RepositoryBuildOrder Include="MusicStore" Order="16" RootPath="$(RepositoryRoot)src\MusicStore\" />
     <RepositoryBuildOrder Include="SignalR" Order="16" />
     <RepositoryBuildOrder Include="AuthSamples" Order="16" RootPath="$(RepositoryRoot)src\AuthSamples\" />
-    <RepositoryBuildOrder Include="Templating" Order="17" RootPath="$(RepositoryRoot)src\Templating\" />
+    <RepositoryBuildOrder Include="Components" Order="17" RootPath="$(RepositoryRoot)src\Components\" />
```
Comment (ryanbrandenburg) [inline build/buildorder.props]: I assume this is here because Blazor relies on SignalR?

**Review by ryanbrandenburg — APPROVED** [review]
LGTM from a structural perspective. Blazor folks will have to opine on if the contents of the files are correct. One question but as long as you've already considered it I'm fine.

File: build/buildorder.props (hunk 1 above)
Comment (SteveSandersonMS) [reply build/buildorder.props]: Correct

File: build/sources.props [hunk 2]
```diff
@@ -19,6 +19,7 @@
       https://dotnet.myget.org/F/aspnetcore-dev/api/v3/index.json;
       https://dotnet.myget.org/F/aspnetcore-tools/api/v3/index.json;
       https://dotnet.myget.org/F/aspnetcore-master/api/v3/index.json;
+      https://dotnet.myget.org/F/blazor-dev/api/v3/index.json;
```
Comment (natemcmaster) [inline build/sources.props]: Out of curiosity, which dependencies did you need from this feed?

File: src/Components/build/arcade.props [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+  -->
+
+  <PropertyGroup>
+    <Configuration Condition=" '$(Configuration)' == '' ">Debug</Configuration>
+    <Platform Condition=" '$(Platform)' == '' ">AnyCPU</Platform>
+    <ArtifactsDir Condition="'$(ArtifactsDir)' == ''">$(RepositoryRoot)artifacts\</ArtifactsDir>
+    <ArtifactsObjDir>$(ArtifactsDir)obj\</ArtifactsObjDir>
+    <ArtifactsConfigurationDir>$(ArtifactsDir)$(Configuration)\</ArtifactsConfigurationDir>
+    <ArtifactsBinDir>$(ArtifactsConfigurationDir)bin\</ArtifactsBinDir>
+    <PackageOutputPath>$(ArtifactsConfigurationDir)packages\</PackageOutputPath>
```
Comment (natemcmaster) [inline src/Components/build/arcade.props]:
```suggestion
    <PackageOutputPath>$(ArtifactsDir)build\</PackageOutputPath>
```

Changing this line should fix the PR build failure.

File: src/Components/build/SignToolData.json [hunk 4]
```diff
@@ -0,0 +1,248 @@
+{
```
Comment (natemcmaster) [inline src/Components/build/SignToolData.json]: Remove this file altogether.

File: src/Components/blazor/version.props [hunk 5]
```diff
@@ -0,0 +1,17 @@
+<Project>
+  <PropertyGroup>
+    <!-- When editing any version prefix here, remember also to update
+         tooling\Microsoft.VisualStudio.BlazorExtension\Properties\AssemblyInfo.cs
+         to include the new version in its upper bounds -->
+    <IsFinalBuild>false</IsFinalBuild>
+    <VersionPrefix>0.8.0</VersionPrefix>
+    <VersionSuffix>preview1</VersionSuffix>
```
Comment (natemcmaster) [inline src/Components/blazor/version.props]:
```suggestion
    <VersionSuffix>preview</VersionSuffix>
```

We have stopped adding X as in "previewX". The guidance is to build as `-preview-$(BuildNumberSuffix)`. See this repo's root version.props file for more details.

File: src/Components/korebuild.json [hunk 6]
```diff
@@ -0,0 +1,14 @@
+{
+  "$schema": "https://raw.githubusercontent.com/aspnet/BuildTools/master/tools/korebuild.schema.json",
+  "channel": "release/2.2",
```
Comment (natemcmaster) [inline src/Components/korebuild.json]:
```suggestion
  "channel": "master",
```

File: src/Components/nuget.config [hunk 7]
```diff
@@ -0,0 +1,11 @@
+<?xml version="1.0" encoding="utf-8"?>
+<configuration>
+  <packageSources>
+    <clear />
+    <add key="nuget.org" value="https://api.nuget.org/v3/index.json" />
```
Comment (natemcmaster) [inline src/Components/nuget.config]: This file should be removed in favor of a src/Components/build/sources.props file. You should be able to copy over https://github.com/aspnet/AspNetCore/blob/master/src/AADIntegration/build/sources.props

**Review by natemcmaster — COMMENTED** [review]
Looks like a great start! The PR builds fail to compile. Let's get that passing before we merge. Here are some things you can tweak that I think will get it working.

File: build/sources.props (hunk 2 above)
Comment (SteveSandersonMS) [reply build/sources.props]: The things that will continue to build in the Blazor repo (JSInterop and Microsoft.AspNetCore.Blazor.Mono)

File: src/Components/build/arcade.props (hunk 3 above)
Comment (SteveSandersonMS) [reply src/Components/build/arcade.props]: Thanks - done

File: src/Components/build/SignToolData.json (hunk 4 above)
Comment (SteveSandersonMS) [reply src/Components/build/SignToolData.json]: Done

File: src/Components/blazor/version.props (hunk 5 above)
Comment (SteveSandersonMS) [reply src/Components/blazor/version.props]: Done

File: src/Components/korebuild.json (hunk 6 above)
Comment (SteveSandersonMS) [reply src/Components/korebuild.json]: Done

**Comment by SteveSandersonMS** [comment]
@natemcmaster It's still failing, and I think it's because `PackageOutputPath` gets passed in by Korebuild as a global property, so it overrides anything configured in the sources inside `src/Components`. What's the intended way to override this?

File: src/Components/build/repo.props [hunk 8]
```diff
@@ -0,0 +1,55 @@
+﻿<Project>
+  <Import Project="dependencies.props" />
+
+  <!-- Properties to make KoreBuild more like dotnet/arcade. -->
+  <PropertyGroup>
+    <BuildDir>$(ArtifactsDir)$(Configuration)\packages\</BuildDir>
```
Comment (natemcmaster) [inline src/Components/build/repo.props]:
```suggestion
```
Aha, here's the other one that should be removed.

File: src/Components/build/repo.props [hunk 9]
```diff
@@ -0,0 +1,55 @@
...
+    <ExcludeFromTest Include="$(RepositoryRoot)test\Microsoft.AspNetCore.Components.E2ETest\Microsoft.AspNetCore.Components.E2ETest.csproj" />
+  </ItemGroup>
+
+  <ItemGroup>
+    <DotNetCoreRuntime Include="$(MicrosoftNETCoreApp20PackageVersion)" />
+    <DotNetCoreRuntime Include="$(MicrosoftNETCoreApp21PackageVersion)" />
+  </ItemGroup>
+
+  <ItemGroup>
+   <AspNetCoreRuntime Include="$(AspNetCoreRuntimeVersion)" />
```
Comment (natemcmaster) [inline src/Components/build/repo.props]: This should also be removed. We cannot install the aspnet runtime before it has been built.

Comment (SteveSandersonMS) [reply src/Components/build/repo.props]: Done

File: src/Components/build/repo.props (hunk 8 above)
Comment (SteveSandersonMS) [reply src/Components/build/repo.props]: That wasn't enough on its own, but tracked it down to needing to add `<OverridePackageOutputPath>false</OverridePackageOutputPath>`. Hopefully will work now - waiting to see.

**Comment by SteveSandersonMS** [comment]
@natemcmaster Alright, it's working now. Unrelated stuff from other parts of the repo still fails like before, but the build, packaging and tests for Components all completes without errors.

OK to merge?

**Review by natemcmaster — APPROVED** [review]

**Comment by natemcmaster** [comment]
To make sure you preserve history, don't use GitHub's squash or rebase merge buttons. Merge via command line or the "create a merge commit" button instead.

**Comment by SteveSandersonMS** [comment]
Will do. I'll do it tomorrow morning my time so I can be around to see the outcome.

**Comment by rynowak** [comment]
Yay! I'm super stoked on this.

**Pull request merged**